### DISCUSSION TITLE
<NuxtLink> only shows a blank page after directing

### DISCUSSION THREAD

**Discussion by isabellaWang0108** [nuxt/nuxt#16433] [Questions] [answered]
I am trying to use `<NuxtLink to="...">` to redirect to a new page. However, my page won't show after the URL change. I have to refresh to render the content. Need help

Here is my Routing code

```
 <template>
      <div class="top-nav-tab">
        <NuxtLink to="/foundations"><h3>Foundations</h3></NuxtLink>
     </div>
   </template>
```
Here is my page code

```
<template>
  <h2>Foundation page</h2>
</template>
```

**Comment by danielroe** [comment, marked as answer]
Likely you have a page that does not have a single root node.

**Reply by isabellaWang0108** [reply]
Will you elaborate on a page's single root node?

**Reply by danielroe** [reply]
It should have a single HTML element or component, not text or several elements. (This is true only for pages and layouts, and is a requirement for transitioning between pages...)

**Reply by isabellaWang0108** [reply]
Gotcha! Problem solved. I did forget to wrap them into a single HTML element. Thank you!

**Reply by OlaAlsaker** [reply]
I had the same issue and thought that comments from Tailwind UI was ignored, but turns out they don't. So if anyone else is experiencing this issue while using Tailwind UI, remember to remove the comments in root 😄 

```vue
<template>
  <!--
    This example requires updating your template:

    ```
    <html class="h-full bg-gray-50">
    <body class="h-full">
    ```
  -->
  <div class="min-h-full flex flex-col justify-center py-12 sm:px-6 lg:px-8">
  </div>
</template>
```

**Reply by c-conklin** [reply]
Can this be effected by any page not having a _single root node_ or just the pages that you redirect to and from?

**Reply by rogadev** [reply]
Interestingly I thought that the pages I was navigating TO had to be in a single-root-element. It turns out my index.vue file was NOT single-root-element and that effected the nuxt routing as well.

Super annoying IMHO... I sure hope they update this in the future to handle regular Vue 3 convention that allows for multiple root elements in a template.

**Reply by danielroe** [reply]
This is a limitation of the Vue `<Transition>` component, which is used for transitions between pages and layouts. We might allow globally disabling transitions, which would enable multiple root elements.

**Reply by rogadev** [reply]
> This is a limitation of the Vue `<Transition>` component, which is used for transitions between pages and layouts. We might allow globally disabling transitions, which would enable multiple root elements.

That actually makes perfect sense. So Nuxt automatically applies a transition component, is that about right? Are there automatic transitions applied or just the wrapping for IF you want to use transitions?

**Reply by danielroe** [reply]
Yes, there is a transition applied: `{ name: 'page', mode: 'out-in' }` and `{ name: 'layout', mode: 'out-in' }` for pages & layouts respectively.

**Reply by rogadev** [reply]
Out of curiosity, what would the worst-case repercussion(s) be if you wrapped multi-root-level-element components into a Nuxt generated single-root-element automatically?

**Reply by danielroe** [reply]
It would be a runtime overhead and potential unexpected behaviour. We are planning to add a warning in dev mode though: https://github.com/nuxt/framework/pull/5469

**Reply by amirnbcode** [reply]
same problem I had, and this is absolutely the right answer

**Reply by leccyril** [reply]
i have the same problem but honestly after readin answer i dont know what to do :/ i have an idex page and form page (and layout with menu) on my index page i redirect to form page, i put a <NuxtLink to="/" /> to redirect on index page, but i have a blank page (only menu is displayed) what i have to do ? what is the purpose of single root node or multiple ? any examples ?

**Reply by chesteralan** [reply]
@leccyril You need to wrap everything inside the **template** tag with a **div** tag
```
<template>
<div>
    <YourComponent />
    <YourComponent />
   <YourComponent />
</div>
</template>
```

**Reply by leccyril** [reply]
oh yes it works well now thanks
